### ISSUE TITLE
Bug: clicking on radio button inside a modal closes the modal

### ISSUE THREAD

**Issue by sbourdon13** [codegouvfr/react-dsfr#165]
Hi,
I'm using RadioButtons inside a modal. If I use it as a controlled component, clicking on a radio button closes the modal.
I reproduced it on stackblitz: https://stackblitz.com/edit/nextjs-lztzqg?file=pages%2Findex.tsx

Now I use radio buttons in uncontrolled mode and it works.

**Comment by garronej** [comment]
Hello @sbourdon13,

Thank you for bringing the issue to our attention.
I suspect that this bug originates from the core library. @lab9fr, could you please review the situation in the playground and confirm my suspicions?

Appreciate your assistance!

**Comment by enguerranws** [comment]
Working on a form with controlled inputs in a Modal component, I thought about this bug (because I didn't encountered it). 

I think I nailed it: @sbourdon13 the issue is that in your example you're instanciating the modal **in** the Home component. So, every render instanciate a new modal component which is closed by default. It should be instanciated **outside** of your component.

I forked you example and it works well : https://stackblitz.com/edit/nextjs-fuaw3d?file=pages%2Findex.tsx
